### PR TITLE
[SPARKR][DOC] regexp_extract should doc that it returns empty string when match fails

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Doc change - see https://issues.apache.org/jira/browse/SPARK-16324
## How was this patch tested?

manual check

### CODE REVIEW

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -2876,7 +2876,8 @@ setMethod("randn", signature(seed = "numeric"),
 
 #' regexp_extract
 #'
-#' Extract a specific(idx) group identified by a java regex, from the specified string column.
+#' Extract a specific \code{idx} group identified by a Java regex, from the specified string column.
```
Comment (shivaram) [inline R/pkg/R/functions.R]: Minor comment: It looks like `idx` was removed in the comment change in https://github.com/apache/spark/pull/14525 -- I'm not sure if it matters, but we could either keep it consistent with the Scala/Python doc or keep the `idx` if you think it helps.

Comment (felixcheung) [inline R/pkg/R/functions.R]: I noticed that - I thought `@param idx a group index` isn't super descriptive on what it is

Comment (shivaram) [inline R/pkg/R/functions.R]: Cool sounds good.

**Comment by shivaram** [comment]
LGTM. Merging into master, branch-2.0

**Pull request closed without merging**